### PR TITLE
Changing the TS importer behavior

### PR DESCRIPTION
**Pull request by kiddinn**
Changing the behavior of the TS importer:

1. Fix an issue with the importer, #276 
2. Add the option to supply custom token passwords
3. Add more error reporting

### CODE REVIEW

**Comment by kiddinn** [comment]
Requires https://github.com/google/timesketch/pull/1288 to be pushed and uploaded to pypi

File: data/recipes/gcp_logging_cloudaudit_ts.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
         "name": "TimesketchExporter",
         "args": {
             "incident_id": "@incident_id",
+	    "token_password": "@token_password",
```
Comment (tomchop) [inline data/recipes/gcp_logging_cloudaudit_ts.json:21]: Plz fix indentation here and in the other json files.

File: data/recipes/gcp_logging_gce_ts.json [hunk 2]
```diff
@@ -26,5 +27,6 @@
         ["start_date", "Start date (yyyy-mm-ddTHH:MM:SSZ)", null],
         ["end_date", "End date (yyyy-mm-ddTHH:MM:SSZ)", null],
         ["--incident_id", "Incident ID (used for Timesketch description)", null],
+        ["--token_password", "Optional custom password to decrypt Timesketch credential file with", ""],
```
Comment (tomchop) [inline data/recipes/gcp_logging_gce_ts.json:30]: Just for consistency, do you mind putting this one after `sketch_id`?

File: dftimewolf/lib/timesketch_utils.py [hunk 3]
```diff
@@ -32,14 +34,23 @@ def GetApiClient(state):
...
-    ts_client = assistant.get_client()
+    config.configure_missing_parameters(
+        config_assistant=assistant, token_password=token_password)
+
+    ts_client = assistant.get_client(token_password=token_password)
+
+    if not ts_client:
+      state.AddError(
+          'Unable to get a Timesketch API Client', critical=False)
+      return ts_client
```
Comment (tomchop) [inline dftimewolf/lib/timesketch_utils.py:45]: return `None` would probably be better here.

File: dftimewolf/lib/timesketch_utils.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 LOCK = threading.Lock()
 
 
-def GetApiClient(state):
+def GetApiClient(state, token_password=''):
```
Comment (tomchop) [inline dftimewolf/lib/timesketch_utils.py:12]: Shouldn't this default to `None`? It feels like encrypting with an empty password should not be the same as not encrypting the file. (unless you're XORing 😅)

File: data/recipes/gcp_logging_cloudaudit_ts.json (hunk 1 above)
Comment (kiddinn) [reply data/recipes/gcp_logging_cloudaudit_ts.json:21]: hmm... the indentation looks "correct" in my editor... let's see

it seems like a `<TAB>` got somehow introduced there in between spaces... fixed

File: dftimewolf/lib/timesketch_utils.py (hunk 3 above)
Comment (kiddinn) [reply dftimewolf/lib/timesketch_utils.py:45]: done

File: dftimewolf/lib/timesketch_utils.py (hunk 4 above)
Comment (kiddinn) [reply dftimewolf/lib/timesketch_utils.py:12]: well, no, the password is a string, and the upstream library expects a string to be passed in, and if the string is empty then the `default` pwd is used, which is auto generated

Comment (tomchop) [reply dftimewolf/lib/timesketch_utils.py:12]: Ack, sounds good. Could you add a bit of docstring that says this?

Comment (kiddinn) [reply dftimewolf/lib/timesketch_utils.py:12]: sure can....

**Pull request merged**